pkg/platform/provider/baremetal/cluster/update.go

```diff
@@ -146,53 +140,6 @@ func (p *Provider) EnsurePreClusterUpgradeHook(ctx context.Context, c *v1.Cluste
 	return util.ExcuteCustomizedHook(ctx, c, platformv1.HookPreClusterUpgrade, c.Spec.Machines[:1])
 }
 
-func (p *Provider) EnsureUpgradeCoreDNS(ctx context.Context, c *v1.Cluster) error {
```

Comment:
these logic shouldnot be removed, you should enhance these logic for 1.21.x case